rclcpp_action/include/rclcpp_action/client.hpp

```diff
@@ -708,23 +720,40 @@ class Client : public ClientBase
     }
     using GoalResultRequest = typename ActionT::Impl::GetResultService::Request;
     auto goal_result_request = std::make_shared<GoalResultRequest>();
+    const GoalUUID goal_id = goal_handle->get_goal_id();
```

Comment:
@jmachowinski @clalancette aside from above discussion, this fixes the another issue. (using weak pointer instead of shared pointer) probably we can proceed this one to address in a part of this issue?

Reply:
@fujitatomoya @clalancette  I openend https://github.com/ros2/rclcpp/pull/2296
to split out the first bug fix.